larpandoracontent/LArCheating/CheatingRemovingCosmicRays.cc

```diff
@@ -0,0 +1,72 @@
+/**
+ *  @file   larpandoracontent/LArCheating/CheatingRemovingCosmicRays.cc
+ * 
+ *  @brief  Implementation of the cheating removing cosmic rays algorithm class.
+ * 
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "Helpers/MCParticleHelper.h"
+
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArCheating/CheatingRemovingCosmicRays.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+CheatingRemovingCosmicRays::CheatingRemovingCosmicRays() :
+    m_inputCaloHitListName(""),
+    m_mcParticleListName("")
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode CheatingRemovingCosmicRays::Run()
+{
+    const MCParticleList *pMCParticleList(nullptr);
+    PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::GetList(*this, m_mcParticleListName, pMCParticleList));
+
+    const CaloHitList *pCaloHitList(nullptr);
+    PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::GetList(*this, m_inputCaloHitListName, pCaloHitList));
+
+    CaloHitList outputCaloHitList;
+
+    for (const CaloHit *pCaloHit : *pCaloHitList)
+    {
+        try
+        {
+            const MCParticle *const pHitParticle(MCParticleHelper::GetMainMCParticle(pCaloHit));
+
+            if (!LArMCParticleHelper::IsCosmicRay(pHitParticle))
+                outputCaloHitList.push_back(pCaloHit);
+        }
+        catch (...)
```

Comment:
Probably wants to catch only `const StatusCodeException &` still, and let anything else go back to algorithm run control.